test/rekt/features/broker/control_plane.go

```diff
@@ -0,0 +1,316 @@
+		state.SetOrFail(ctx, t, TriggerNameKey, triggerName)
+	})
+
+	f.Stable("Conformance").
+		Should("Triggers SHOULD include a Ready condition in their status.",
+			triggerHasReadyInConditions).
+		Should("The Trigger SHOULD indicate Ready=True when events can be delivered to its subscriber.",
+			readyTriggerCanDeliver).
+		Should("While a Trigger is Ready, it SHOULD indicate its subscriber's URI via the `status.subscriberUri` field.",
+			readyTriggerHasSubscriberURI).
+		Must("Triggers MUST be assigned to exactly one Broker.",
+			triggerHasOneBroker).
+		Must("The assigned Broker of a Trigger SHOULD be immutable.",
+			triggerSpecBrokerIsImmutable)
+
+	return f
+}
+
+func ControlPlaneTrigger_WithBrokerLifecycle() *feature.Feature {
+	f := feature.NewFeatureNamed("Trigger, With Broker Lifecycle")
+
+	subscriberName := feature.MakeRandomK8sName("sub")
+	f.Setup("Install Subscriber", svc.Install(subscriberName, "bad", "svc"))
+
+	triggerName := feature.MakeRandomK8sName("trigger")
+	f.Setup("Create a Trigger", trigger.Install(triggerName, "",
+		trigger.WithSubscriber(svc.AsRef(subscriberName), ""),
+	))
+
+	f.Setup("Set Trigger Name", func(ctx context.Context, t feature.T) {
+		state.SetOrFail(ctx, t, TriggerNameKey, triggerName)
+	})
+
+	f.Stable("Conformance").
+		Should("Triggers SHOULD be assigned a default Broker upon creation if no Broker is specified by the user.",
+			todo).
+		May("A Trigger MAY be created before its assigned Broker exists.",
+			todo).
+		Should("A Trigger SHOULD progress to Ready when its assigned Broker exists and is Ready.",
+			todo)
+	return f
+}
+
+func ControlPlaneTrigger_WithValidFilters(brokerName string) *feature.Feature {
+	f := feature.NewFeatureNamed("Trigger, With Filters")
+	f.Setup("Set Broker Name", setBrokerName(brokerName))
+
+	subscriberName := feature.MakeRandomK8sName("sub")
+	f.Setup("Install Subscriber", svc.Install(subscriberName, "bad", "svc"))
+
+	// CloudEvents attribute names MUST consist of lower-case letters ('a' to 'z') or digits ('0' to '9') from the ASCII character set. Attribute names SHOULD be descriptive and terse and SHOULD NOT exceed 20 characters in length.
+	filters := map[string]string{
+		"source":               "a source",
+		"id":                   "an id",
+		"specversion":          "the spec version",
+		"type":                 "the type",
+		"subject":              "a subject",
+		"time":                 "a time",
+		"datacontenttype":      "a datacontenttype",
+		"dataschema":           "a dataschema",
+		"aaa":                  "bbb",
+		"c1d2e3":               "123",
+		"abcdefghijklmnopqrst": "max length",
+	}
+
+	triggerName := feature.MakeRandomK8sName("trigger")
+	f.Setup("Create a Trigger", trigger.Install(triggerName, brokerName,
+		trigger.WithSubscriber(svc.AsRef(subscriberName), ""),
+		trigger.WithFilter(filters),
+	))
+
+	f.Setup("Set Trigger Name", func(ctx context.Context, t feature.T) {
+		state.SetOrFail(ctx, t, TriggerNameKey, triggerName)
```

Comment:
You don't need this, you can just pass the name to `getTrigger`. This way is more convoluted and hard to follow

Reply:
disagree. I can now break apart the tests from the setup, which is the goal

Reply:
Ok i see your point, then the `Set` should be inside the install phase, not here.

And you still need to consider that a test may install multiple triggers

Reply:
no, the install phase is a StepFn so I need to pass data to it and then set the data. This keeps install flexible and the domain specific tests that know there is a trigger in the settings can pull it. But install does not require it.